commit 5c8ae642eb0443b68a14716d3c27d66b3e71b705
Author: Gilles Caulier <caulier.gilles@gmail.com>
Date:   Thu Jun 16 17:40:04 2011 +0200

    add libusb as linking dependency for gphoto2 library to prevent crash at libgphoto2 init when OpenCV library is linked against libdc1394.
    BUGS: 268267

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0dd8b04..7d21a57 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -205,7 +205,7 @@ IF (ENABLE_INTERNALMYSQL)
                         /opt/mysql/libexec /usr/mysql/bin /opt/mysql/bin ${MYSQL_TOOLS_PATH}
                        ENV MYSQL_TOOLS_PATH )
     MESSAGE(STATUS "Found MySQL install_db executable at: ${INITCMD_MYSQL}")
-    SET(DATABASESERVERDIR "${CMAKE_CURRENT_SOURCE_DIR}/databaseserver")    
+    SET(DATABASESERVERDIR "${CMAKE_CURRENT_SOURCE_DIR}/databaseserver")
 ELSE (ENABLE_INTERNALMYSQL)
     SET(DATABASESERVERDIR "")
 ENDIF (ENABLE_INTERNALMYSQL)
@@ -238,6 +238,11 @@ ENDIF (KMAP_VERSION)
 MACRO_OPTIONAL_FIND_PACKAGE(Gphoto2)
 MACRO_BOOL_TO_01(GPHOTO2_FOUND HAVE_GPHOTO2)
 
+IF (NOT WIN32 AND HAVE_GPHOTO2)
+    # See B.K.O #268267 : digiKam need to be linked to libusb to prevent crash at gphoto2 init if opencv is linked with libdc1394
+    MACRO_OPTIONAL_FIND_PACKAGE(USB)
+ENDIF (NOT WIN32 AND HAVE_GPHOTO2)
+
 MACRO_OPTIONAL_FIND_PACKAGE(KdepimLibs)
 MACRO_BOOL_TO_01(KDEPIMLIBS_FOUND HAVE_KDEPIMLIBS)
 
@@ -340,17 +345,17 @@ IF(GPHOTO2_FOUND)
     EXEC_PROGRAM(gphoto2-config ARGS --version RETURN_VALUE _return_VALUE OUTPUT_VARIABLE GPHOTO2_VERSION)
     STRING(REPLACE "libgphoto2" "" GPHOTO2_VERSION "${GPHOTO2_VERSION}")
     MACRO_ENSURE_VERSION("2.4.0" "${GPHOTO2_VERSION}" VERSION_GPHOTO2)
-    IF(VERSION_GPHOTO2)
+    IF(VERSION_GPHOTO2 AND LIBUSB_FOUND)
         SET(GPHOTO2_FOUND true)
-    ELSE(VERSION_GPHOTO2)
+    ELSE(VERSION_GPHOTO2 AND LIBUSB_FOUND)
         SET(GPHOTO2_FOUND false)
-    ENDIF(VERSION_GPHOTO2)
+    ENDIF(VERSION_GPHOTO2 AND LIBUSB_FOUND)
 ENDIF(GPHOTO2_FOUND)
 
 IF(GPHOTO2_FOUND)
-    MESSAGE(STATUS " libgphoto2 library found................. YES (optional)")
+    MESSAGE(STATUS " libgphoto2 and libusb libraries found.... YES (optional)")
 ELSE(GPHOTO2_FOUND)
-    MESSAGE(STATUS " libgphoto2 library found................. NO  (optional)")
+    MESSAGE(STATUS " libgphoto2 and libusb libraries found.... NO  (optional)")
     MESSAGE(STATUS "")
     MESSAGE(STATUS " digiKam will be compiled without GPhoto2 support.")
     MESSAGE(STATUS "")
--- a/core/CMakeLists.txt
+++ b/core/CMakeLists.txt
@@ -281,6 +281,12 @@ TARGET_LINK_LIBRARIES(digikam
         cameragui
 )
 
+IF (NOT WIN32 AND GPHOTO2_FOUND)
+    # See B.K.O #268267 : digiKam need to be linked to libusb to prevent crash at gphoto2 init if opencv is linked with libdc1394
+    TARGET_LINK_LIBRARIES(digikam ${LIBUSB_LIBRARIES})
+ENDIF (NOT WIN32 AND GPHOTO2_FOUND)
+
+
 IF(USE_SCRIPT_IFACE)
     TARGET_LINK_LIBRARIES(digikam ${QT_QTSCRIPT_LIBRARY})
 ENDIF(USE_SCRIPT_IFACE)
